PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/layers/transmitter.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+#include "../layers/layers_wrapper.h"
+#include "./globals.h"
+#include "../utils/queue_wrapper.h"
+#include "../utils/dllp.h"
+#include "../utils/tlp.h"
+#include "../layers/flit.h"
+#include <vector>
+#include <queue>
+#include <iterator>
+
+/**
+ * @brief Represents a transmitter that sends packets over a network.
+ *
+ * This class encapsulates the functionality required to transmit packets over
+ * a network using a credit-based flow control mechanism. It maintains a set of
+ * global parameters that govern the behavior of the transmitter, as well as a
+ * set of wrapper classes that provide access to the various layers of the
+ * network protocol.
+ */
+class Transmitter {
+public:
+	Globals globals;
+	LayersWrapper* layers;
+	QueueWrapper<TLP*>* listenOn;
```

Comment:
I think the transmitter doesn't need a listenOn queue; since the thread function listens and passes the packets. Make sure and if so remove it

Reply:
yeah, u r 👉